src/lightning_app/cli/lightning_cli.py

```diff
@@ -26,13 +34,149 @@ def get_app_url(runtime_type: RuntimeType, *args) -> str:
+@click.option("--provider", "provider", type=str, default="aws", help="cloud provider to be used for your cluster")
+@click.option("--external-id", "external_id", type=str, required=True)
+@click.option(
+    "--role-arn", "role_arn", type=str, required=True, help="AWS role ARN attached to`the associated resources."
+)
+@click.option(
+    "--region",
+    "region",
+    type=str,
+    required=False,
+    default="us-east-1",
+    help="AWS region which is used to host the associated resources.",
+)
+@click.option(
+    "--instance-types",
+    "instance_types",
+    type=str,
+    required=False,
+    default=",".join(default_instance_types),
+    help="Instance types which you desire to support for computer jobs within the cluster.",
+)
+@click.option(
+    "--cost-savings",
+    "cost_savings",
+    type=bool,
+    required=False,
+    default=False,
+    is_flag=True,
+    help="using this flag ensures that the cluster is created with a profile that is optimized for "
+    "cost saving, making runs cheaper but start-up times may increase",
+)
+@click.option(
+    "--edit-before-creation",
+    default=False,
+    is_flag=True,
+    help="Edit the created cluster spec before submitting to API server.",
+)
+@click.option(
+    "--wait",
+    "wait",
+    type=bool,
+    required=False,
+    default=False,
+    is_flag=True,
+    help="using this flag CLI will wait until the cluster is running",
+)
+def create_cluster(
+    cluster_name: str,
+    region: str,
+    role_arn: str,
+    external_id: str,
+    provider: str,
+    instance_types: str,
+    edit_before_creation: bool,
+    cost_savings: bool,
+    wait: bool,
+    **kwargs,
+):
+    """Create a Lightning.ai compute cluster with provided cloud provider credentials."""
+    if provider != "aws":
+        click.echo("only provider aws is supported today")
+        return
+    aws = AWSClusterManager()
+    aws.create(
+        cluster_name=cluster_name,
+        region=region,
+        role_arn=role_arn,
+        external_id=external_id,
+        instance_types=instance_types.split(","),
+        edit_before_creation=edit_before_creation,
+        cost_savings=cost_savings,
+        wait=wait,
+    )
+
+
+@clusters.command("list")
+def list_clusters(**kwargs):
+    """List your Lightning.ai compute clusters."""
+    mgr = AWSClusterManager()
+    mgr.list()
+    pass
+
+
+@clusters.command("delete")
+@click.argument("cluster", type=str)
+@click.option(
+    "--force",
+    "force",
+    type=bool,
+    required=False,
+    default=False,
+    is_flag=True,
+    help="Force delete cluster from grid system. This does NOT delete any resources created by the cluster, "
+    "just cleaning up the entry from the grid system. You should not use this under normal circumstances",
+)
+@click.option(
+    "--wait",
+    "wait",
+    type=bool,
+    required=False,
+    default=False,
+    is_flag=True,
+    help="using this flag CLI will wait until the cluster is deleted",
```

Comment:
Please use: 

Enabling this flag makes the CLI wait until the cluster is deleted.